### PR TITLE
FIX : #6747

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: htdocs/core/class/html.form.class.php [hunk 1]
```diff
@@ -1435,8 +1435,8 @@ function select_dolusers($selected='', $htmlname='userid', $show_empty=0, $exclu
        {
         	if (! empty($conf->multicompany->transverse_mode))
         	{
-        		$sql.= ", ".MAIN_DB_PREFIX."usergroup_user as ug";
-        		$sql.= " WHERE ug.fk_user = u.rowid";
+        		$sql.= " LEFT JOIN ".MAIN_DB_PREFIX."usergroup_user as ug";
+        		$sql.= " ON ug.fk_user = u.rowid";
```
Comment (hregis) [inline htdocs/core/class/html.form.class.php]: Why ?
and missing WHERE with this !

Comment (eldy) [reply htdocs/core/class/html.form.class.php]: Sure. It miss a WHERE 1 = 1 to make next AND later working.
@regis I think problem is when you remove all groups of a user and in transverse mode.  It disappears from all user lists. Am i right @atm-florian ?

Comment (ghost) [reply htdocs/core/class/html.form.class.php]: @eldy Exact. This is the point if a user is not in a group it not selectable in select in transverse mode. Just update the PR

**Comment by eldy** [comment]
@hregis Are you ok now with the fix of @atm-florian ? Should address the situation where user has no groups at all and must be visible into combo list I think.

**Comment by hregis** [comment]
@atm-florian @eldy 
i see no difference !! Where is the problem ?

File: htdocs/core/class/html.form.class.php (hunk 1 above)
Comment (hregis) [reply htdocs/core/class/html.form.class.php]: @atm-florian @eldy 
i try with this fix but i see no difference !
and if a user is not in a group/entity he can not selectable in this entity !

Comment (hregis) [reply htdocs/core/class/html.form.class.php]: @eldy @atm-florian 
with this code is ok, i see all users :
but a user must be in group/entity to be selectable !!

$sql.= " LEFT JOIN ".MAIN_DB_PREFIX."usergroup_user as ug";
$sql.= " ON ug.fk_user = u.rowid";
$sql.= " AND ug.entity = ".$conf->entity;
$sql.= " WHERE 1 = 1";

**Comment by ghost** [comment]
@hregis $sql.= ", ".MAIN_DB_PREFIX."usergroup_user as ug WHERE ug.fk_user = u.rowid"; will exclude user without group but LEFT JOIN will be OK to display them

**Comment by hregis** [comment]
@atm-florian user must be in group/entity to be selectable !!

**Pull request merged**